kafka-connect-mqtt/src/main/scala/com/datamountaineer/streamreactor/connect/mqtt/source/MqttManager.scala

```diff
@@ -100,7 +100,7 @@ class MqttManager(connectionFn: MqttSourceSettings => MqttConnectOptions,
           val scalaList: Seq[String] = l
           scalaList
         }.getOrElse(Seq.empty[String])
-        Option(converter.convert(kafkaTopic, topic, message.getId.toString, message.getPayload, keys, kcql.getKeyDelimeter)) match {
+        Option(converter.convert(kafkaTopic, kcql.getSource, message.getId.toString, message.getPayload, keys, kcql.getKeyDelimeter)) match {
```

Comment:
Why would this be better? If non-wildcard subscriptions are used the MQTT topic and the kcql.getSource are the same.

However, on wildcard subscriptions you might not want to provide the expression used ans lose the actual topic name.